contributors/TEMPLATING.md

```diff
@@ -176,9 +176,19 @@ The special files and folders are:
 
 ## Merging package.json files
 
-The package we use to merge package.json files [merge-packages](3) will use the last version of a dependency given a conflict. For example:
+The package we use to merge `package.json` files [merge-packages](3) will attempt to find intersections of dependencies or use the latest version of a dependency in case of a conflict.
+
+For example:
 
 ```
+version on file one: ~1.2.3
+version on file two: ^1.0.0
+resulting version: ~1.2.3
+
+version on file one: ^1.0.0
+version on file two: >=1.2.0 <1.3.0
+resulting version: ~1.2.0
+
 version on file one: 1.0.0
 version on file two: 0.1.0
 resulting version: 0.1.0
```

Comment:
«will attempt to find intersections of dependencies or use the latest version of a dependency in case of a conflict.»

This is not happening here in this example, right? It's using the second file version, even if it's an older version.

Should we remove this example? Or we need to tweak the explanation?

Reply:
> This is not happening here in this example, right? It's using the second file version, even if it's an older version.
> Should we remove this example? Or do we need to tweak the explanation?

Since there are exact versions, there are no intersections. Hence examples are right and I think we need them because they show how last version was taken

---

I also tried

version on file one: ^1.0.0
version on file two: ^0.1.0
And looks like result should be ^1.0.0, but
resulting version is ^0.1.0

For reversed
version on file one: ^0.1.0
version on file two: ^1.0.0
resulting version is ^1.0.0

I think the reason is that major versions are different => there can be breaking changes => no intersections.

Another examples
version on file one: ^1.1.0
version on file two: ^1.2.0
resulting version: ^1.2.0

For reversed
version on file one: ^1.3.0
version on file two: ^1.2.0
resulting version is ^1.3.0

And here intersections works as expected.

I think we don't need to add all these examples to readme, it's just to understand how it works

Reply:
Ok, I see. It wasn't clear to me reading this

> The package we use to merge `package.json` files [merge-packages](3) will attempt to find intersections of dependencies or use the latest version of a dependency in case of a conflict.

Maybe we could expand a bit?

> The package we use to merge `package.json` files [merge-packages](3) will attempt to find intersections of the dependencies or use the latest version of a dependency in case of a conflict. If there is no intersection, it'd use the version of the file computed last.

up you... it's confusing anyway haha

Reply:
[30825c1](https://github.com/scaffold-eth/create-eth/pull/79/commits/30825c17c368c5040ccd223a9ccd7c0d64a5b52e)

I think now it's better